src/inference/src/model_reader.cpp

```diff
@@ -210,7 +226,7 @@ std::shared_ptr<ov::Model> read_model(const std::shared_ptr<AlignedBuffer>& mode
 
     FE = manager.load_by_model(params);
     if (FE) {
-        FE->add_extension(ov_exts);
+        FE->add_extension(filter_extensions_for_fronend(FE, ov_exts));
```

Comment:
`FE->add_extension` can be called outside of `Core` class, so we need a solution which does not assume `Core`

BTW, see comment in CVS-160510

Reply:
I added extension check into frontend instead of core